nixos/modules/security/pam.nix

```diff
@@ -719,6 +746,14 @@ in
 
   config = {
 
+    assertions = [
+      { assertion = !config.security.pam.enableSSHAgentAuth;
+        message = ''
+          This option has been deprecated due to a security vunerability. Read
+          the 19.09 release notes for more informations and the replacing options.
+        ''; }
+    ];
```

Comment:
You could remove `options.security.pam.enableSSHAgentAuth` and use `mkRemovedOptionModule` with the text of this assertion instead. Less code. :)

Reply:
I wish I could but I need `assertion` because it `throw`, `mkRemovedOptionModule` only add a warnings which doesn't:
https://github.com/NixOS/nixpkgs/blob/f71206190fd462d6d592ad846f8de2e3047d5026/lib/modules.nix#L592-L602
I don't want someone to be locked out of their systems because of the breaking change.

Reply:
I see, thanks!

Reply:
Since https://github.com/NixOS/nixpkgs/pull/69419 `mkRemovedOptionModule` throws, so it should be usable for this now

Reply:
Since https://github.com/NixOS/nixpkgs/pull/69419 `mkRemovedOptionModule` throws, so it should be usable for this now